library/ssl_tls13_client.c

```diff
@@ -2488,6 +2524,18 @@ static int ssl_tls13_parse_new_session_ticket_exts( mbedtls_ssl_context *ssl,
         {
             case MBEDTLS_TLS_EXT_EARLY_DATA:
                 MBEDTLS_SSL_DEBUG_MSG( 4, ( "early_data extension received" ) );
+                if( extension_data_len != 4 )
+                {
+                    MBEDTLS_SSL_PEND_FATAL_ALERT(
+                        MBEDTLS_SSL_ALERT_MSG_DECODE_ERROR,
+                        MBEDTLS_ERR_SSL_DECODE_ERROR );
+                    return( MBEDTLS_ERR_SSL_DECODE_ERROR );
+                }
+                if( ssl->session != NULL )
+                {
+                    ssl->session->ticket_flags |=
+                            MBEDTLS_SSL_TLS1_3_TICKET_ALLOW_EARLY_DATA;
+                }
```

Comment:
Sorry for this comments, I just realized， this is in NewSessionTicket, it has different structure. I do not think it belongs this PR. 

At this moment, please set MBEDTLS_SSL_TLS1_3_TICKET_ALLOW_EARLY_DATA when received NewSessionTicket.

GnuTLS client will send early data without `early_data extension` . Above suggestion is to follow up GnuTLS.

I think we should think about this behavior in next PRs. 

```suggestion
```

Reply:
Per meeting conclusion, close it